src/dwarf/debug_info.rs

```diff
@@ -0,0 +1,1325 @@
+//! Parse the `.debug_info` section to get Debug Information Entries.
+//!
+//! It supports DWARFv4 now. (See https://dwarfstd.org/doc/DWARF4.pdf)
+//! It parse DIEs from the `.debug_info` section and Abbreviations
+//! from the `.debg_abbrev` section.
+//!
+//! The `.debug_info` section is a list of (Compile-)Units.  Every
+//! Unit comprises DIEs to carry debug information of a source file.
+//! A Unit starts with a header to describe the size of this unit in
+//! the section, the offset of its abbreviation table in the
+//! `.debug_abbrev` section, ..., and DWARF version. (version 4)
+//!
+//! A DIE starts with an index encoded in LEB128 to the abbreviation
+//! table of the Unit.  The abbreviation given by the index describle
+//! the content, format and layout of a DIE.  The abbreviation index
+//! starts from 1.  0 means a null entry.  DIEs in an Unit are
+//! organized as a tree, parent-children.  Null entries are used to
+//! signal the last sibling to end a level.
+//!
+//! A user starts a parser by creating an instance of [`UnitIter`].
+//! It will walk through the data in the `.debug_info` and
+//! `.debug_abbrev` section to return Units.
+use super::constants::*;
+use super::{decode_leb128_128, decode_leb128_128_s, decode_udword, decode_uhalf, decode_uword};
+use crate::tools::extract_string;
+use std::io::{Error, ErrorKind};
+use std::iter::Iterator;
+use std::mem;
+
+fn decode_3bytes_usigned(data: &[u8]) -> u32 {
+    data[0] as u32 | ((data[1] as u32) << 8) | ((data[2] as u32) << 16)
+}
+
+#[allow(dead_code)]
```

Comment:
is there some better way than marking every single struct or function as `#allow[dead_code]`? What other libraries do when they have some code that might be compiled out due to disabled features? I'd expect there is some cleaner way?

Reply:
the question is whether Rust really requires marking crates or functions as dead code if some functionality is gated by a feature? I'd imagine there must be some better solution, as that seems to be a pretty common thing across Rust libraries. Can you please check what other libraries do in such cases?